internal/test_cases/history_file_assertion.go

```diff
@@ -0,0 +1,48 @@
+// Package test_cases provides helpers for shell test assertions.
+package test_cases
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/codecrafters-io/shell-tester/internal/utils"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// AssertFileHasCommandsInOrder reads the file at filePath, splits it into lines, and checks that each line matches the Command field of the corresponding CommandResponseTestCase.
+// It logs a success message for each line and returns a detailed error if any line does not match or if the line count is wrong.
+func AssertFileHasCommandsInOrder(l *logger.Logger, filePath string, commands []string) error {
```

Comment:
```suggestion
func AssertHistoryFileHasCommands(l *logger.Logger, filePath string, commands []string) error {
```

This is in `test_cases`, so isn't clear that a "file" is a "history file". The "in order" part I think is obvious so could remove (or keep, I don't mind either way thought this would be less verbose)